mssqlcli/jsonrpc/tests/test_jsonrpcclient.py

```diff
@@ -188,11 +189,27 @@ def test_receive_invalid_response_exception(self):
             self.assertEqual(
                 str(exception),
                 u'Content-Length was not found in headers received.')
+
+            # flaky test fix: logic that waits until threads shut down
+            count = 0
+            while count < 5:
+                active_threads = threading.enumerate()
+                if test_client.request_thread not in active_threads or \
+                    test_client.response_thread in active_threads:
+                    time.sleep(5)
+                    count += 1
```

Comment:
what if we pass 5 and thread is still not shutdown?

Reply:
Then the test will likely fail, because the following assertion checks for a request thread that has likely not been created.

The only way the request thread _could_ have been created following 5 failed retries is if it happens to come alive following the fifth try. I have not seen this happen during my testing.